CIP-0129/README.md

```diff
@@ -0,0 +1,170 @@
+Type - `Key Hash`
+
+Identifier - `0200000000000000000000000000000000000000000000000000000000`
+
+Bech32 - `cc_hot1qgqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqvcdjk7`
+
+#### Constitutional Committee Cold Credential
+
+Key - `00000000000000000000000000000000000000000000000000000000`
+Type - `Script Hash`
+
+Identifier - `1300000000000000000000000000000000000000000000000000000000`
+
+Bech32 - `cc_cold1zvqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq6kflvs`
+
+#### DRep Credential
+
+Key - `00000000000000000000000000000000000000000000000000000000`
+Type - `Key Hash`
+
+Identifier - `2200000000000000000000000000000000000000000000000000000000`
+
+Bech32 - `drep1ygqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqq7vlc9n`
+
+## Rationale: how does this CIP achieve its goals?
+
+By introducing a header and payload design for different keys, we can achieve specifying credentials in a hex format (bytes), being able to share and use across tooling is possible. We also specify bech32 prefixes for user friendly identification of the identifiers.
+
+
+## Path to Active
+
+### Acceptance Criteria
+Tools, Wallets, and Explorers to utilize the identifiers and bech32 prefixes defined in this CIP for communication and view purposes.
+
+- [ ] Requires updating Ledger nano app, and Trezor. The changes can be proposed once the CIP is merged.
+- [ ] Tooling
+  - [x] CNTools
+  - [x] SPO Scripts
+  - [x] typhonjs
+  - [x] Gov Tools
+  - [x] cardano-signer
+- [ ] APIs
+  - [x] Koios
+  - [x] Cardanoscan API
+  - [ ] Blockfrost
+- [x] Explorers
+  - [x] Cardanoscan.io
+  - [x] AdaStat.net
+- [ ] Wallets
+  - [x] Eternl
+  - [x] Typhon Wallet
+  - [ ] Lace
+
+### Implementation Plan
+- This CIP uses some bech32 prefixes which are already defined by CIP105 and ref by CIP005, This PR includes updates to both the CIPs with updated vector spec. The suggested changes align with the design of the original CIP005.
+
+- For key generation tools - To move faster to this CIP, current tools which implement CIP105 based bech32 prefixes do not have to implement this CIP specification. They can change the prefix according to the updates suggested in this CIP, updating bech32 prefix will be a find and replace sort of an updated and it will instantly become compatible with this CIP.
+
+- Tools like explorers and wallets - These tools can potentially support both formats to start with for the purpose of allowing users to search for drep, cold keys, hot keys as these 3 are impacted by this CIP upgrade. And can continue to show only this CIP specification, having an easy backward compatibility as well moving to this CIP standard.
+
+- This CIP does not require a hard fork for implementation, the goal is to use the identifies specified in this CIP for the UI, and as a medium of communication for sharing such keys and IDs (though it will be good if this CIP is also enforced at the ledger level, simplifying CDDL and bringing data storage efficiency).
```

Comment:
Please remove this statement, since that would be a major breaking change and it would require a whole new CIP. Most importantly, I believe, it would a bad idea to implement something like this in ledger because:
* contrary to what it says, it would reduce storage efficiency since every credential would require one more byte in serialized form for the header and we have a lot of credentials on chain. There is a way to modify the existing one byte tag that we use for this in order to not incur new overhead, but there is definitely no way to improve storage efficiency when it comes to credentials.
* Changing serialization for an already existing type would have to span a few ledger eras and would result in a lot of tools being broken, **including all hardware wallets**.
* Such change is not needed for ledger, so we should not affect ledger semantics.
* There is no benefit in such change for the on chain representation or ledger users.
* bech32, prefixes or other credential identifiers are only useful for end users and tooling that faces them, so CDDL and ledger should be kept out of it. There is nothing wrong, from the ledger perspective, for users using the same cryptographic keys or same scripts for multiple entities.
```suggestion
- This CIP does not require a hard fork for implementation, the goal is to use the identifies specified in this CIP for the UI, and as a medium of communication for sharing such keys and IDs.
```

Reply:
i agree, no need to change it in the ledger storage. this is a frontend "issue" not a ledger issue.

Reply:
updated